.github/workflows/check.yml

```diff
@@ -4,23 +4,23 @@ jobs:
   check:
     strategy:
       matrix:
-        os:
-          [
-            '{"runs-on": "ubuntu-latest"}',
-            '{"runs-on": "macos-latest"}',
-            '{"runs-on": "ubuntu-latest", "container": "ubuntu:18.04"}',
-            '{"runs-on": "windows-latest"}',
-          ]
-    runs-on: ${{ fromJSON(matrix.os).runs-on }}
-    container: ${{ fromJSON(matrix.os).container }}
+        system:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        include:
+          - system: ubuntu-18.04
+            container: ubuntu:18.04    
```

Comment:
does this not work?

```suggestion
        - ubuntu-18.04
```

Reply:
Yes, it would, but the original intent was to test it inside a container (?)

Reply:
outdated